middleware/reverse/README.md

```diff
@@ -0,0 +1,65 @@
+# reverse
+
+The *reverse* middleware allows CoreDNS to respond dynamic to an PTR request and the related A/AAAA request.
+
+## Syntax
+
+~~~
+reverse NETWORK {
+    [template TEMPLATE]
+    [domain DOMAIN]
+    [ttl TTL]    
+}
+~~~
+
+* **NETWORK** is the CIDR formatted network to respond on.
+* `template` inject the ip to an template for the hostname. Defaults to "ip-{ip}".
+    v4 replaces the "." with an "-". 10.1.1.1 results in "10-1-1-1"
+    v6 removes the ":" and fills the zeros. "ffff::ffff" results in "ffff000000000000000000000000ffff" 
+* `domain` defaults to the listen zone.
+* `ttl` defaults to 60
+
+Note that you have to make sure that this middleware will get actual queries for the
+following zones: `.in-addr.arpa.`, `.ip6.arpa.`.
+
+## Examples
+
+~~~
+# Serve on port 53
+# match arpa. and compute.internal. to resolv reverse and forward lookup 
+.:53 {
+    # proxy unmatched requests
+    proxy 8.8.8.8
+    
+    # answer requests for IPs in this network 
+    # PTR 1.0.32.10.in-addr.arpa. 3600 ip-10-0-32-1.compute.internal.
+    # A ip-10-0-32-1.compute.internal. 3600 10.0.32.1
+    reverse 10.32.0.0/16 {
+        # template of the ip injection to hostname
+        template "ip-{ip}"
+        
+        # the domain to match forward lookup and answer to 
+        domain "compute.internal."
+    
+        # set time-to-live of the RR
+        ttl 3600
+    }
+
+    # v6 is also possible
+    # PTR 1.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.1.0.d.f.ip6.arpa. 3600 ip-fd010000000000000000000000000001.compute.internal.
+    # AAAA ip-fd010000000000000000000000000001.compute.internal. 3600 fd01::1  
+    reverse fd01::/16 {
+        # template of the ip injection to hostname
+        template "ip-{ip}"
+        
+        # the domain to match forward lookup and answer to 
+        domain "compute.internal."
```

Comment:
The domain is more an dns suffix to the hostname. Its added to the templated hostname on an PTR request (since we need an FQDN). On forward A/AAAA requests we would like to get our ip back, so we check if we would like to answer for this domain/zone and its part of our template. 

The regex was pretty fast while testing, but i thought myself about storing the precompiled regex.